apps/web/src/api/hooks/useUploadToStorage.ts

```diff
@@ -0,0 +1,42 @@
+import axios from 'axios';
+import { useMutation, UseMutationOptions } from '@tanstack/react-query';
+import { IResponseError, MIME_TYPE_TO_FILE_EXTENSION, UploadTypesEnum } from '@novu/shared';
+
+import { useGetSignedUrl } from './useGetSignedUrl';
+
+export const useUploadToStorage = (
+  options: UseMutationOptions<string, IResponseError, { file: File; type: UploadTypesEnum }> = {}
+) => {
+  const { getSignedUrl } = useGetSignedUrl();
+  const { mutateAsync: uploadToStorage, ...mutationData } = useMutation(
+    async ({ file, type }) => {
+      const { signedUrl, path, additionalHeaders } = await getSignedUrl({
+        extension: MIME_TYPE_TO_FILE_EXTENSION[file.type],
+        type,
+      });
+      await axios.put(signedUrl, file, {
+        headers: {
+          'Content-Type': file.type,
+          ...additionalHeaders,
+        },
+        transformRequest: [
+          (data, headers) => {
+            if (headers) {
+              // eslint-disable-next-line
+              delete headers.Authorization;
+            }
+
+            return data;
+          },
+        ],
+      });
+
+      return path;
+    },
+    {
+      ...options,
+    }
+  );
+
+  return { uploadToStorage, ...mutationData };
+};
```

Comment:
when the `async` function is passed to the `useMutation` from the `@tanstack/react-query` library you don't need to handle the thrown exceptions, it will be handled internally by the `useMutation` hook. You can pass the options object as a second argument to the `useMutation` hook with `onError` prop that will be called when the error is thrown in the `async` function.